In [1]:
import pandas as pd

nsmes1988_df = pd.read_csv("NSMES1988-NSMES1988.csv")

nsmes1988_df

nsmes1988_df.info(verbose=True, show_counts=True)
nsmes1988_df.describe(include='all')
na_rows = nsmes1988_df.isna().any(axis=1)
nsmes1988_df.loc[na_rows]
#print(nsmes1988_df.value_counts()[nsmes1988_df.value_counts().dtype == 'object'])

nsmes1988_df.to_json("nsmes1988.json")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4406 entries, 0 to 4405
Data columns (total 19 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  4406 non-null   int64  
 1   visits      4406 non-null   int64  
 2   nvisits     4406 non-null   int64  
 3   ovisits     4406 non-null   int64  
 4   novisits    4406 non-null   int64  
 5   emergency   4406 non-null   int64  
 6   hospital    4406 non-null   int64  
 7   health      4406 non-null   object 
 8   chronic     4406 non-null   int64  
 9   adl         4406 non-null   object 
 10  region      4406 non-null   object 
 11  age         4406 non-null   float64
 12  gender      4406 non-null   object 
 13  married     4406 non-null   object 
 14  school      4406 non-null   int64  
 15  income      4406 non-null   float64
 16  employed    4406 non-null   object 
 17  insurance   4406 non-null   object 
 18  medicaid    4406 non-null   object 
dtypes: float64(2), int64(9), ob

# NSMES Data
####  Null Data:
        Contains no null values that I could find off the bat. The range was 0 - 4405 and each row reported 4406 non-null values.
#### Column Names:
        Many names are less descriptive than they could be. As characters in a header, it isn't too expensive to give them better names at the expense of header readability on printout.
        visits  -> num_physician_office_visits
        nvisits -> num_non_physician_office_visits
        ovisits -> num_outpatient_physician_office_visits
        novisits-> num_outpatient_non_physician_office_visits
        emergency -> num_emergency_visits
        hospital -> num_hospital_stays
        health -> self_reported_health_factor
        chronic -> num_chronic_conds
        adl -> daily_acty_limit_factor
        region -> region
        age -> age_div_10
        gender -> gender
        married -> married
        school -> num_school_years
        income -> income_div_10k
        employed -> employed
        insurance -> has_priv_insurance
        medicaid -> has_medicaid_coverage

many dtypes could be updated and or changed to be boolean (e.g married, employed) and others could be enumerated to save on string space in larger databases. Just use a key

